### ISSUE TITLE
Fix FP S4043 ('no-misleading-array-reverse'): Consider optional chaining

### ISSUE THREAD

**Issue by saberduck** [SonarSource/SonarJS#2513]
User reported under RSPEC https://jira.sonarsource.com/browse/RSPEC-4043 

Hi, I tried this in SonarJS, it seems `items?.sort()` reports while `items.sort()` does not which is unexpected.

What means this rule is not compatible with `optional chaining`.

**Comment by victor-diez-sonarsource** [comment]
@ilia-kebets-sonarsource @yassin-kammoun-sonarsource can you reproduce this? both `items?.sort()` and `items.sort()` report in my case. I would say this was probably fixed some time ago?

**Comment by yassin-kammoun-sonarsource** [comment]
@victor-diez-sonarsource Here is a snippet that highlights the false positive:

```javascript
const xs = [1, 2, 3]; 
xs.sort();  // OK
xs?.sort(); // FP
```

**Comment by victor-diez-sonarsource** [comment]
oh, the FP is without assignment. ok, got it! thanks

**Comment by yassin-kammoun-sonarsource** [comment]
Indeed. Quoting the RSPEC of S4043:

> This rule raises an issue when the return values of these methods are assigned, which could lead maintainers to overlook the fact that the original value is altered.
